miden-crypto/src/hash/sha2/mod.rs

```diff
@@ -0,0 +1,448 @@
+//! SHA2 hash function wrappers (SHA-256 and SHA-512).
+
+use alloc::string::String;
+use core::{
+    mem::size_of,
+    ops::Deref,
+    slice::{self, from_raw_parts},
+};
+
+use sha2::Digest as Sha2Digest;
+
+use super::{Digest, ElementHasher, Felt, FieldElement, Hasher, HasherExt};
+use crate::utils::{
+    ByteReader, ByteWriter, Deserializable, DeserializationError, HexParseError, Serializable,
+    bytes_to_hex_string, hex_to_bytes,
+};
+
+#[cfg(test)]
+mod tests;
+
+// CONSTANTS
+// ================================================================================================
+
+const DIGEST256_BYTES: usize = 32;
+const DIGEST512_BYTES: usize = 64;
+
+// SHA256 DIGEST
+// ================================================================================================
+
+/// SHA-256 digest (32 bytes).
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "serde", serde(into = "String", try_from = "&str"))]
+pub struct Sha256Digest([u8; DIGEST256_BYTES]);
```

Comment:
I wonder if it would be cleaner to have `struct Sha2Digest<const D: usize>([u8; D])` with type alias `Sha256Digest = Sha2<256>`.

Reply:
I think it might be worth having a more general `Digest256` and `Digest512` struct which can be reused among the different binary hash functions. But let's create an issue and do this in a follow up